app/assets/javascripts/trim_link.js

```diff
@@ -3,9 +3,47 @@
 
 var trimLink = {};
 
+
 // Prepare the callback on trim upload controls
 trimLink.trimFileUpload = function() {
   'use strict';
+  var $fileField = $('input[type=file]');
+
+  if ($('#pq-details').length) {
+    $('#status').hide();
+    $('#choices').on('click', function () {
+      $('#pq_trim_link_attributes_file').click();
+    });
+    $('#cancel').on('click', function () {
+      $('form.progress-menu-form').trigger('reset');
+      $('#status').hide();
+      $('#choices').show();
+    });
+
+    // File selector behaviour
+    $fileField.on('change', function(){
+      var chosen = $fileField.val();
+      var selectedFileName;
+      var $messageContainer = $('#status');
+      // var $messageIcon = $('.fa');
+      var $uploadMessage = $('#tr5-message');
+      var $chooseBtn = $('#choices');
+      if(chosen) {
+        // If a file is selected successfully, show a message
+        selectedFileName = chosen.split(/[\\/]/).pop();
+        $chooseBtn.hide();
+        $uploadMessage.text('File selected: ' + selectedFileName);
+        $messageContainer.show();
+      } else {
+        //or else, go back to showing just the 'Choose Trim link' button
+        $chooseButton.show();
```

Comment:
commented out the else section and tested, seems to work fine, recommend Nick to test and ensure the build succeeds, and Travis is happy